common/djangoapps/track/utils.py

```diff
@@ -28,3 +30,38 @@ def default(self, obj):  # pylint: disable=method-hidden
             return obj.isoformat()
 
         return super(DateTimeJSONEncoder, self).default(obj)
+
+
+class LatinJSONEncoder(DateTimeJSONEncoder):
+    """
+    JSON encoder for UnicodeDecodeError with latin1 characters
+    """
+
+    def encode(self, obj):
+        """
+        encode method gets an original object
+        and returns result string. obj argument will be the
+        object that is passed to json.dumps function
+        """
+        obj = self.iterate_dictionary(obj)
+
+        return super(LatinJSONEncoder, self).encode(obj)
+
+    def iterate_dictionary(self, obj):
+        """
+        Iterate over nested dict structure
+        """
+        for key, value in obj.iteritems():
+            if isinstance(value, dict):
+                self.iterate_dictionary(value)
+            elif isinstance(value, str):
```

Comment:
Oh, good point.  I failed to recognize that iterate_dictionary *assumes* that it is receiving a list, not receiving an iterable.  So that's actually a shortcoming of the encode() method -- it's calling iterate_dictionary because it's assuming that it's being called with a dictionary.  And I don't think you can make that assumption. 

But what I wrote about handling lists is clearly wrong:  it cannot call self.iterate_dictionary on the values.  You need something more general to traverse the object's contents.  (And hopefully there are no loops. :)